### PR TITLE
feat: implementing the basics of blockchain

### PR DESCRIPTION
**Pull request by Ilagouilly**
Applying the following fundamentals of Blockchain, that are:

* Creation of blocks that store data.
* Generation of a digital signature that chains blocks together.
* Validation of new blocks by providing proof of work.
* Verification that Blocks are valid and data is unchanged.

### CODE REVIEW

File: src/main/java/com/cryptocurrency/newbchain/Block.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@ToString
+@Getter
+@Setter
+public class Block {
+    
+    private String hash;
+	private String previousHash;
+	private String data; // data will be contain a summary of a transaction.
+	private long timeStamp; 
+    private int nonce;
```
Comment (NevilClavain) [inline src/main/java/com/cryptocurrency/newbchain/Block.java:20]: More comment on it plz

File: src/main/java/com/cryptocurrency/newbchain/Block.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	public Block(String data, String previousHash) {
+		this.data = data;
+		this.previousHash = previousHash;
+		this.timeStamp = new Date().getTime();
+        this.hash = calculateBlockHash();
+	}
+
+    public String calculateBlockHash() {
```
Comment (NevilClavain) [inline src/main/java/com/cryptocurrency/newbchain/Block.java:29]: Maybe this method can be private

File: src/main/java/com/cryptocurrency/newbchain/Block.java (hunk 1 above)
Comment (Ilagouilly) [reply src/main/java/com/cryptocurrency/newbchain/Block.java]: done bitmoji

File: src/main/java/com/cryptocurrency/newbchain/Block.java (hunk 2 above)
Comment (Ilagouilly) [reply src/main/java/com/cryptocurrency/newbchain/Block.java]: done & done ;)

Comment (Ilagouilly) [reply src/main/java/com/cryptocurrency/newbchain/Block.java]: Actually, the method is used in Main class to check the validity of the Blockchain. However, I reduced its visibility to package-private ;)

Comment (NevilClavain) [reply src/main/java/com/cryptocurrency/newbchain/Block.java]: ok, fine for me

**Review by NevilClavain — APPROVED** [review]

**Pull request merged**